File: core/FindAllNumbersDisappearedInAnArray/FindAllNumbersDisappearedInAnArray.java
package com.dbc;

import java.util.ArrayList;
import java.util.List;

public class FindAllNumbersDisappearedInAnArray {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList<>();
        for (int num : nums) {
            num = Math.abs(num);
            nums[num - 1] = -Math.abs(nums[num - 1]);
        }
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) res.add(i + 1);
        }
        return res;
    }
}

File: core/FindAllNumbersDisappearedInAnArray/FindAllNumbersDisappearedInAnArray.py
from typing import List


def findDisappearedNumbers(nums: List[int]) -> List[int]:
    res, length = [], len(nums)
    for num in nums:
        num = abs(num)
        nums[num - 1] = -abs(nums[num - 1])

    for index in range(length):
        if nums[index] > 0:
            res.append(index + 1)

    return res



